File: SUMFOUR.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	int n, Count = 0;
	cin >> n;
	vector<long> a(n), b(n), c(n), d(n);
	for (int i = 0; i < n; i++)
		cin >> a[i] >> b[i] >> c[i] >> d[i];
	vector<long> x(n*n), y(n*n);
	for (int i = 0; i < n; i++)
		for (int j = 0; j < n; j++)
			x[Count] = a[i] + b[j], y[Count++] = c[i] + d[j];
	sort(y.begin(), y.end());
	long long ans = 0;
	for (int i = 0; i < Count; i++)
		ans += (upper_bound(y.begin(), y.end(), -x[i]) - lower_bound(y.begin(), y.end(), -x[i]));
	cout << ans << endl;
}

File: SUMFOUR.py
# will get TLE

import bisect as bs

n, ans = int(input()), 0
A = []
B = []
C = []
D = []
for i in range(n):
	a, b, c, d = map(int, input().split())
	A.append(a)
	B.append(b)
	C.append(c)
	D.append(d)
X = [a+b for a in A for b in B]
Y = [c+d for c in C for d in D]
Y.sort()
for x in X:
	ans += (bs.bisect_right(Y, -x) - bs.bisect_left(Y, -x))
print(ans)
